From 61c88eef1ca48726c58e1d88a78c30492e020fb3 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 18 Nov 2024 13:28:54 +0200
Subject: [PATCH] [SW-208080] gaudi3: incorporate new fit load type for erom

New fit load type for erom is introduced to help agent code to
identify, this fit contains erom binaries.

So this patch incorporated the FIT_LOAD_TYPE_FLASH_EROM into
agent-fw_loader_erom.its

Change-Id: I7be5b60027e217dc05ecef81f5ad7f4083c11719
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/board/habanalabs/gaudi3_arc/agent-fw_loader_erom.its b/board/habanalabs/gaudi3_arc/agent-fw_loader_erom.its
index ddce288..9c20c24 100644
--- a/board/habanalabs/gaudi3_arc/agent-fw_loader_erom.its
+++ b/board/habanalabs/gaudi3_arc/agent-fw_loader_erom.its
@@ -64,7 +64,7 @@
 		default = "fw-loader-conf";
 
 		fw-loader-conf {
-			load-type = <4>; /* FIT_LOAD_TYPE_FLASH */
+			load-type = <@FIT_LOAD_TYPE_FLASH_EROM@>;
 			description = "Habanalabs device boot loader";
 			loadables = "erom_pcie", "erom_recovery", "img-ptr";
 		};
diff --git a/board/habanalabs/gaudi3_arc/post-image-agents.sh b/board/habanalabs/gaudi3_arc/post-image-agents.sh
index 06dbc47..f11c6cb 100644
--- a/board/habanalabs/gaudi3_arc/post-image-agents.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-agents.sh
@@ -15,6 +15,7 @@
 GAUDI3_SRAM_ADDR_LO=0xF0000000
 GAUDI3_SRAM_ADDR_HI=0x0200FFFF
 AGENT_BASE_ADDR=$((GAUDI3_SRAM_ADDR_HI << 32 | GAUDI3_SRAM_ADDR_LO))
+FIT_LOAD_TYPE_FLASH_EROM=$(grep -E "FIT_LOAD_TYPE_FLASH_EROM[^_]" ${SPECS_EMB_PROJ_DIR}/fit_load_type.h | sed 's/.* = \(.*\),/\1/')
 IMG_PTR_SPI_OFFSET=0x00060000
 EROM_IMG_PTR_OFFSET=0x00000000
 SECURITY_SPI_OFFSET=0xFFFFFFFF
@@ -42,6 +43,7 @@
 ITS_SED_PARAMS+="s/@EROM_PCIE_FLASH_OFFSET@/${EROM_PCIE_OFFSET}/g;"
 ITS_SED_PARAMS+="s/@EROM_RECOVERY_FLASH_OFFSET@/${EROM_RECOVERY_OFFSET}/g;"
 ITS_SED_PARAMS+="s/@EROM_IMG_PTR_FLASH_OFFSET@/${EROM_IMG_PTR_OFFSET}/g;"
+ITS_SED_PARAMS+="s/@FIT_LOAD_TYPE_FLASH_EROM@/${FIT_LOAD_TYPE_FLASH_EROM}/g;"
 
 AGENT_NIC_FW_FILE=${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_NAME}-eth-phy.bin.le
 AGENT_SERDES_FW_NAME=zephyr_agentfw_serdes.bin
